ocs_ci/utility/utils.py

```diff
@@ -692,22 +690,30 @@ def get_openshift_client(
             (default: config.RUN['client_version'])
         bin_dir (str): Path to bin directory (default: config.RUN['bin_dir'])
         force_download (bool): Force client download even if already present
+        skip_comparison (bool): Skip the comparison between the existing OCP client
+            version and the configured one.
 
     Returns:
         str: Path to the client binary
 
     """
     version = version or config.RUN["client_version"]
-    version = expose_ocp_version(version)
     bin_dir = os.path.expanduser(bin_dir or config.RUN["bin_dir"])
     client_binary_path = os.path.join(bin_dir, "oc")
     kubectl_binary_path = os.path.join(bin_dir, "kubectl")
     download_client = True
     client_version = None
+    try:
+        version = expose_ocp_version(version)
+    except AssertionError:
```

Comment:
What about to have general Exception here? In the case the service will be in bad shape and will not return the proper json but some other output we will hit the same issue. 

I see that that get_url_content is raising the AssertionError but then we do json.loads so it can error also there.
So I am for make it more robust with handle Exception, we do log warning. Maybe you can print the exception message here to have all the details written in that warning.

Reply:
I have broadened to `Exception` here as well as changed from logging a warning to logging the original exception that caused this to fail.